assignment/js/main.js

```diff
@@ -0,0 +1,199 @@
+    highlightFeature(zipgon)
+
+    // Get zip code
+    zipcode = zipgon.feature.properties.CODE;
+    console.log(zipcode)
+    $('#zipcode').text(zipcode);
+    $('#pop').text(popObj[zipcode])
+
+    // vaccination data
+    var fullyVcc = vccData[zipcode]["fully_vaccinated"]
+    var partiallyVcc = vccData[zipcode]["partially_vaccinated"]
+
+    // bar chart
+    Chart.defaults.font.size = 14;
+    $('#barChart').remove();
+    $('#bar').append('<canvas id="barChart"></canvas>')
+    var ctx = document.getElementById('barChart').getContext('2d');
+    var barChart = new Chart(ctx, {
+        type: 'bar',
+        data: {
+            labels: ['fully', 'partially'],
+            datasets: [{
+                label: '# of vaccinated People',
+                data: [fullyVcc, partiallyVcc],
+                backgroundColor: [
+                    'rgba(227,26,28,0.7)',
+                    'rgba(253,141,60,0.7)'
+                ],
+                borderColor: [
+                    'rgb(128,0,38)',
+                    'rgb(227,26,28)'
+                ],
+                borderWidth: 1
+            }]
+        },
+        options: {               
+            scales: {
+                y: {
+                    beginAtZero: true
+                }
+            }
+
+        }
+    });
+
+    // Pie chart
+    var vccPeo = fullyVcc + partiallyVcc;
+    var nonVccPeo = popObj[zipcode] - vccPeo;
+    console.log(nonVccPeo)
+
+    $('#pieChart').remove();
+    $('#pie').append('<canvas id="pieChart"></canvas>')
+    ctx = document.getElementById('pieChart').getContext('2d');
+    if(pieChart) {pieChart.destroy()}
+    var pieChart = new Chart(ctx, {
+        type: 'pie',
+        data: {
+            labels: ['vaccinated', "not vaccinated"],
+            datasets: [{
+                data: [vccPeo, nonVccPeo],
+                backgroundColor: [
+                    'rgba(253,141,60,0.7)',
+                    'rgba(255, 237, 160,0.7)'
+                  ],
+                hoverOffset: 4 
+            }]      
+        }
+    })
+}
+
+function onEachFeature(feature, layer) {
+    layer.on({
+        click: zipgonClick
+    });
+}
+
+$.ajax('https://raw.githubusercontent.com/CPLN692-MUSA611-Open-Source-GIS/datasets/master/geojson/Zipcodes_Poly.geojson').done(function(data) {
+    var parsedData = JSON.parse(data);
+    console.log(parsedData)
+
+    $.ajax('https://raw.githubusercontent.com/CPLN692-MUSA611-Open-Source-GIS/OSGIS-week9/master/assignment/vaccination_by_zip.json').done(function(data) {
+        vccData = JSON.parse(data);
+        // console.log(vccData);
+        $('#date').text(vccData[Object.keys(vccData)[0]]['etl_timestamp'].substr(0,10));
+
+        var url = 'http://www.whateverorigin.org/get?url=' + encodeURIComponent("http://zipatlas.com/us/pa/philadelphia/zip-code-comparison/population-density.htm") + '&callback=?';
+        
+        $.getJSON(url, function(data){
+            // Crawl population by zipcode
```

Comment:
This is 'callback hell'. I'll be discussing strategies for getting around it in class today but suffice it to say that working with/around it is no small task.